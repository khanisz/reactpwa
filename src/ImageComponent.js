import React from "react";
import PropTypes from "prop-types";
import { withStyles } from "@material-ui/core/styles";
import Card from "@material-ui/core/Card";
import CardActionArea from "@material-ui/core/CardActionArea";
import CardContent from "@material-ui/core/CardContent";
import CardMedia from "@material-ui/core/CardMedia";
import Typography from "@material-ui/core/Typography";

const styles = {
  card: {
    maxWidth: 450,
    margin: 10
  },
  media: {
    height: 300
  }
};

function ImageComponent(props) {
  const { classes } = props;
  const { title, text, url } = props;
  return (
    <div>
      {/* <div style={{ height: 10 }} /> */}
      <Card className={classes.card}>
        <CardActionArea>
          <CardMedia
            className={classes.media}
            image={url}
            title="Contemplative Reptile"
          />
          <CardContent>
            <Typography gutterBottom variant="h5" component="h2">
              {title}
            </Typography>
            <Typography component="p">{text}</Typography>
          </CardContent>
        </CardActionArea>
      </Card>
    </div>
  );
}

ImageComponent.propTypes = {
  classes: PropTypes.object.isRequired
};

export default withStyles(styles)(ImageComponent);
