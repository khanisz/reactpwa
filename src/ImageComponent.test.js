import React from "react";
import { shallow } from "enzyme";
import ImageComponent from "./ImageComponent";

describe("ImageComponent", () => {
  it('should render correctly in "debug" mode', () => {
    const component = shallow(<ImageComponent debug />);
    expect(component).toMatchSnapshot();
  });

  it("should render correctly with no props", () => {
    const component = shallow(<ImageComponent />);
    expect(component).toMatchSnapshot();
  });

  it("should render correctly with title", () => {
    const title = "Tytul";
    const component = shallow(<ImageComponent title={title} />);
    expect(component).toMatchSnapshot();
  });

  it("should render correctly with text", () => {
    const text = "Lore ipsum";
    const component = shallow(<ImageComponent text={text} />);
    expect(component).toMatchSnapshot();
  });

  it("should render correctly with title and text", () => {
    const title = "Tytul";
    const text = "Lore ipsum";
    const component = shallow(<ImageComponent title={title} text={text} />);
    expect(component).toMatchSnapshot();
  });

  it("should render correctly with title, text, url", () => {
    const title = "Tytul";
    const text = "Lore ipsum";
    const url ='url';
    const component = shallow(<ImageComponent title={title} text={text} url={url} />);
    expect(component).toMatchSnapshot();
  });


});
