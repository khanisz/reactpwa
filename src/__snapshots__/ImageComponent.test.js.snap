// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImageComponent should render correctly in "debug" mode 1`] = `
<ImageComponent
  classes={
    Object {
      "card": "ImageComponent-card-1",
      "media": "ImageComponent-media-2",
    }
  }
  debug={true}
/>
`;

exports[`ImageComponent should render correctly with no props 1`] = `
<ImageComponent
  classes={
    Object {
      "card": "ImageComponent-card-1",
      "media": "ImageComponent-media-2",
    }
  }
/>
`;

exports[`ImageComponent should render correctly with text 1`] = `
<ImageComponent
  classes={
    Object {
      "card": "ImageComponent-card-1",
      "media": "ImageComponent-media-2",
    }
  }
  text="Lore ipsum"
/>
`;

exports[`ImageComponent should render correctly with title 1`] = `
<ImageComponent
  classes={
    Object {
      "card": "ImageComponent-card-1",
      "media": "ImageComponent-media-2",
    }
  }
  title="Tytul"
/>
`;

exports[`ImageComponent should render correctly with title and text 1`] = `
<ImageComponent
  classes={
    Object {
      "card": "ImageComponent-card-1",
      "media": "ImageComponent-media-2",
    }
  }
  text="Lore ipsum"
  title="Tytul"
/>
`;

exports[`ImageComponent should render correctly with title, text, url 1`] = `
<ImageComponent
  classes={
    Object {
      "card": "ImageComponent-card-1",
      "media": "ImageComponent-media-2",
    }
  }
  text="Lore ipsum"
  title="Tytul"
  url="url"
/>
`;
